mmdeploy/codebase/mmdet/models/dense_heads/condinst_head.py

```diff
@@ -0,0 +1,359 @@
+            batched_param_preds.new_zeros(N, 1, batched_param_preds.shape[2]),
+        ),
+        1,
+    )
+    batched_points = torch.cat((batched_points, batched_points.new_zeros((N, 1, 2))), 1)
+    batched_strides = torch.cat((batched_strides, batched_strides.new_zeros((N, 1))), 1)
+
+    # sort
+    is_use_topk = keep_top_k > 0 and (
+        torch.onnx.is_in_onnx_export() or keep_top_k < batched_dets.shape[1]
+    )
+    if is_use_topk:
+        _, topk_inds = batched_dets[:, :, -1].topk(keep_top_k, dim=1)
+    else:
+        _, topk_inds = batched_dets[:, :, -1].sort(dim=1, descending=True)
+    topk_batch_inds = torch.arange(
+        batch_size, dtype=topk_inds.dtype, device=topk_inds.device
+    ).view(-1, 1)
+
+    batched_dets = batched_dets.index_select(0, topk_batch_inds[0]).index_select(1, topk_inds[0])
+    batched_labels = batched_labels.index_select(0, topk_batch_inds[0]).index_select(1, topk_inds[0])
+    batched_param_preds = batched_param_preds.index_select(0, topk_batch_inds[0]).index_select(1, topk_inds[0])
+    batched_points = batched_points.index_select(0, topk_batch_inds[0]).index_select(1, topk_inds[0])
+    batched_strides = batched_strides.index_select(0, topk_batch_inds[0]).index_select(1, topk_inds[0])
+    # slice and recover the tensor
+    return (
+        batched_dets,
+        batched_labels,
+        batched_param_preds,
+        batched_points,
+        batched_strides,
+    )
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    "mmdet.models.dense_heads.CondInstMaskHead.forward_single"
+)
+def condinst_mask_head__forward_single(
+    self, mask_feat: Tensor, positive_info: InstanceData
+):
+    """Rewrite function about forwarding features of a each image."""
+    pos_param_preds = positive_info.get("param_preds")
+    pos_points = positive_info.get("points")
+    pos_strides = positive_info.get("strides")
+
+    num_inst = pos_param_preds.shape[0]
+    mask_feat = mask_feat[None].repeat(num_inst, 1, 1, 1)
+    _, _, H, W = mask_feat.size()
+    if num_inst == 0:
+        return (pos_param_preds.new_zeros((0, 1, H, W)),)
+
+    locations = self.prior_generator.single_level_grid_priors(
+        mask_feat.size()[2:], 0, device=mask_feat.device
+    )
+
+    rel_coords = relative_coordinate_maps(
+        locations, pos_points, pos_strides, self.size_of_interest, mask_feat.size()[2:]
+    )
+    mask_head_inputs = torch.cat([rel_coords, mask_feat], dim=1)
+
+    weights, biases = _parse_dynamic_params(self, pos_param_preds)
+    mask_preds = _dynamic_conv_forward(
+        self, mask_head_inputs, weights, biases, num_inst
+    )
+    mask_preds = mask_preds.reshape(-1, H, W)
+    mask_preds = aligned_bilinear(
+        mask_preds.unsqueeze(0), int(self.mask_feat_stride / self.mask_out_stride)
+    ).squeeze(0)
+
+    return (mask_preds,)
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    "mmdet.models.dense_heads.CondInstMaskHead.predict_by_feat"
+)
+def condinst_mask_head__predict_by_feat(
```

Comment:
we should remove rewritings `condinst_mask_head__forward_single` and put everything in this funciton